mesa/visualization/ModularVisualization.py

```diff
@@ -246,6 +251,10 @@ def on_message(self, message):
                 else:
                     self.application.model_kwargs[param] = value
 
+        elif msg_type == "shut_down":
```

Comment:
Why is it necessary to be in debug mode in order for the shutdown to be functional?

Reply:
My way of thoughts was:
* It's generally not a good idea to allow a client to shut down a web server.
* In a local environment, in debug mode, this might be OK.

By the way, I have a general question:
Do I understand correctly that this Tornado server is expected to run only in a local environment, on the user's computer?
Or is it also deployed somewhere as an actual web server? :thinking:

Reply:
> Or is it also deployed somewhere as an actual web server? :thinking:

Some people might have done this, that I am not aware of. The relevant issue is #481. We are also currently exploring the possibility of having the UI stack fully in Python (https://github.com/projectmesa/mesa/discussions/1622), in a way that can be run on a Jupyter notebook. Hosting Jupyter notebooks is so ubiquitous that there are canned ways to do it securely.

Reply:
I am in the opinion that if the pure Python implementation goes well, it might replace the current clunky Tornado server, in a way that is easy to extend, and based on existing established data interaction frameworks. cc: @ankitk50.

Another past attempt at revamping the frontend: #1334 by @Corvince.